### PR TITLE
DOC: stats: explain rice scaling; add a note to the tutorial about scaling

### PR DESCRIPTION
**Pull request by efiring**
While in the vicinity I made a few other edits, and there are a few trailing whitespace deletions. This documention suggestion is related to #4559 and #4538, and follows on my ill-conceived and closed #4605. It is largely an attempt to prevent a future scipy user from getting as confused as I was.

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -3779,6 +3778,12 @@ class rice_gen(rv_continuous):
 
     `rice` takes ``b`` as a shape parameter.
 
+    The Rice distribution describes the length, `r`, of a 2-D vector
+    with components `(U+u, V+v)`, where `U, V` are constant, and `u, v`
+    are independent Gaussian random variables with standard deviation
+    `s`.  Let `R = (U**2 + V**2)**0.5`. Then the pdf of `r` is
+    `rice.pdf(x, R/s, scale=s)`.
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Typesetting nitpick: I think this needs double backticks for ReST layout (double backticks for code, single backticks for cross-references). Or explicit `:math:` blocks.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -3792,6 +3797,9 @@ def _rvs(self, b):
         return np.sqrt((t*t).sum(axis=0))
 
     def _pdf(self, x, b):
+        # For numerical stability, use (x**2 + b**2)/2 = ((x-b)**2)/2 + xb.
+        # The factor of exp(-xb) is then included in the i0e function
+        # in place of the modified Bessel function, i0.
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Strictly speaking, it's the other way around: we use `i0e` for stability, and compensate for the extra exponential by rearranging the quadratic form. But this is really minor.

**Comment by ev-br** [comment]
+111 from me modulo a typesetting nitpick.

**Comment by efiring** [comment]
I decided the double backtick formatting was a better compromise for the docstring than filling it with :math: blocks.  I also reworded the method comment to clarify the issue with numerics. I squashed and force-pushed, so there is still only a single commit.

**Comment by ev-br** [comment]
Merging, thanks @efiring !

**Pull request merged**